src/mscorlib/shared/System/Text/ValueStringBuilder.cs

```diff
@@ -21,42 +21,42 @@ public ValueStringBuilder(Span<char> initialBuffer)
             _pos = 0;
         }
 
-        public string GetString()
+        public int Length => _pos;
+
+        public override string ToString()
         {
             var s = new string(_chars.Slice(0, _pos));
-
-            char[] toReturn = _arrayToReturnToPool;
-            this = default; // for safety, to avoid using pooled array if this instance is erroneously appended to again
-
-            if (toReturn != null)
-            {
-                ArrayPool<char>.Shared.Return(toReturn);
-            }
-
+            Clear();
             return s;
         }
 
         public bool TryCopyTo(Span<char> destination, out int charsWritten)
         {
-            if (_pos > destination.Length)
+            if (_chars.Slice(0, _pos).TryCopyTo(destination))
+            {
+                charsWritten = _pos;
+                Clear();
+                return true;
+            }
+            else
             {
                 charsWritten = 0;
+                Clear();
                 return false;
             }
+        }
 
-            bool copied = _chars.Slice(0, _pos).TryCopyTo(destination);
-            Debug.Assert(copied);
-            charsWritten = _pos;
-
-            char[] toReturn = _arrayToReturnToPool;
-            this = default; // for safety, to avoid using pooled array if this instance is erroneously appended to again
-
-            if (toReturn != null)
+        public void Insert(int index, char value, int count)
+        {
+            if (_pos > _chars.Length - count)
             {
-                ArrayPool<char>.Shared.Return(toReturn);
+                Grow(count);
             }
 
-            return true;
+            int remaining = _pos - index;
+            _chars.Slice(index, remaining).CopyTo(_chars.Slice(index + count));
```

Comment:
Can you please explain how we are we sure that `_chars.Length > index + count` so that `_chars.Slice(index + count)` will succeed?

Reply:
This is an internal helper. The caller needs to ensure their arguments are correct. This explicitly is not doing argument validation.

Reply:
Where it this helper being used? I can't seem to find any place that calls it.

Reply:
It's not used in coreclr right now. The file is "shared" (at the moment by copy/paste) with corefx, where it's used by BigInteger.